### PR TITLE
New Exchange: tokenomy

### PR DESCRIPTION
**Pull request by yogisofiharja**

### CODE REVIEW

File: js/tokenomy.js [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+                        'openOrders',
+                        'cancelOrder',
+                        'orderHistory',
+                        'withdrawCoin',
+                    ],
+                },
+            },
+            'markets': {
+                // HARDCODING IS DEPRECATED
+                // but they don't have a corresponding endpoint in their API
```
[the remark this answers is not in the document]
Comment (yogisofiharja) [reply js/tokenomy.js]: Thanks for your review.
I'm a little bit confused about "we" in "we have to remove ....". We means you or me?

[the remark this answers is not in the document]
Comment (kroitor) [reply js/tokenomy.js]: @yogisofiharja well, we usually spend some time to polish the code before we merge it, so, when i say *we*, I mostly ask you, but i can't demand anything from you, so, if you want to help us clear it up – that would be great and your involvement is very welcome. If not – we (me, other people from the community) will try to do that as soon as possible. So, *we* means *either you or me, whoever does it first, and if no one does it before I have the time to fix it myself – I will*.

[the remark this answers is not in the document]
Comment (yogisofiharja) [reply js/tokenomy.js]: Hi @kroitor , finally I can make a new pull request regarding this Tokenomy exchange platform. I hope you can review it soon. Thank you

**Comment by yogisofiharja** [comment]
@kroitor can you confirm that this PR is ok? thank you

**Comment by kroitor** [comment]
@yogisofiharja 

> @kroitor can you confirm that this PR is ok?

Nope, unfortunately, I cannot confirm that. If it was ok, we would have no reasons not to merge it. In fact, if a PR is ok – it is usually merged right away. And the remaining open and hanging PRs you see – those are not ok for this or that reason. So, a good confirmation would be to merge it. However, if this PR is not merged, this means it's not ok, unfortunately.

Most often the reason is violating our CONTRIBUTING guidelines: 

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes

Most importantly, you're using symbols, market ids and currency ids interchangeably, which is a violation of this very important unification rule:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#sending-market-ids
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#parsing-symbols

Apart from those, several other rules are violated throughout the implementation file.

We keep it open, because we hope to do the cleanup and merge it as soon as we can. If you can go through the guidelines and add the necessary fixes – that would really help speed it up. Thanks!

**Comment by yogisofiharja** [comment]
@kroitor 
Thank you for your feedback. Will work on it soon.
Thanks!

**Comment by brandsimon** [comment]
@kroitor 
`And the remaining open and hanging PRs you see – those are not ok for this or that reason`
What is the problem with @5465? :p

**Comment by kroitor** [comment]
- https://github.com/ccxt/ccxt/pull/5465/files#diff-0c5d6793e3bd19d2d629bf86650e194aR372
_Sends symbol as the 2nd argument to `parseOrder` where `market` is expected_

- https://github.com/ccxt/ccxt/pull/5465/files#diff-0c5d6793e3bd19d2d629bf86650e194aR334 _Breaks array length rule multiple times here https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#working-with-array-lengths_

- https://github.com/ccxt/ccxt/pull/5465/files#diff-0c5d6793e3bd19d2d629bf86650e194aR133 _Breaks https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#parsing-symbols_

- https://github.com/ccxt/ccxt/pull/5465/files#diff-0c5d6793e3bd19d2d629bf86650e194aR173 _Breaks https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#parsing-symbols_

- https://github.com/ccxt/ccxt/pull/5465/files#diff-0c5d6793e3bd19d2d629bf86650e194aR217 _Returns array instead of tickers indexed by symbol_

- https://github.com/ccxt/ccxt/pull/5465/files#diff-0c5d6793e3bd19d2d629bf86650e194aR236
_Ignores `this.timeframes` map (see how other exchanges do this)_

- https://github.com/ccxt/ccxt/pull/5465/files#diff-0c5d6793e3bd19d2d629bf86650e194aR268
_Breaks https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings_

I may have missed some other minor issues.

**Comment by ttodua** [comment]
dead exchange

**Pull request closed without merging**